util/design/secded_gen.py

```diff
@@ -317,6 +322,53 @@ def verify(cfgs):
     return error
 
 
+def ecc_encode(codetype: str, k: int, dataword: int) -> Tuple[int, int]:
+    log.info(f"Encoding ECC for {hex(dataword)}")
+
+    # first check to see if bit width is supported among configuration
+    with open(SECDED_CFG_PATH, 'r') as infile:
+        config = hjson.load(infile)
+
+    codes = None
+    bitmasks = None
+    m = None
+    for cfg in config['cfgs']:
+        if cfg['k'] == k and cfg['code_type'] == codetype:
+            m = cfg['m']
+            codes = gen_code(codetype, k, m)
+            bitmasks = calc_bitmasks(k, m, codes, False)
+            break
+
+    # error if k not supported
+    if not m:
+        raise Exception(f'ECC for length {k} of type {codetype} unsupported')
+
+    # represent supplied dataword as a binary string
+    word_bin = format(dataword, '0' + str(k) + 'b')
+
+    codeword = word_bin
+    for j, mask in enumerate(bitmasks):
```

Comment:
if `j` isn't used, maybe change to `for mask in bitmasks:`?

Reply:
o yes good point, that `j` was a previous edition.